src/background/services/wallet/WalletService.ts

```diff
@@ -450,6 +456,36 @@ export class WalletService implements OnUnlock {
     }
   }
 
+  async signTransactionBatch(
+    batch: TransactionRequest[],
+    network: Network,
+    tabId?: number,
+  ) {
+    const wallet = await this.getWallet({ network, tabId });
+
+    if (!wallet) {
+      throw new Error('Wallet not found');
+    }
+
+    // Only wallets that provide us signed transactions without additional approvals
+    // can be used to sign transaction batches (so for example hardware wallets or accounts
+    // that connect through WalletConnect protocol would not work, since users have to approve
+    // each transaction one by one anyways.
```

Comment:
I think we should add a follow up story and make the other other wallets capable of something similar. 
i.e: After the user presses approve all, the transactions would display one after the other on the hw wallet without UI interaction. The Ledger dialog could display,  "Approve transaction 1/2 ... (and thisplay the details)"

Reply:
👌 -> https://ava-labs.atlassian.net/browse/CP-9818